example/oven_fsm.dart

```diff
@@ -0,0 +1,226 @@
+            counterReset < 1,
+            en < 0,
+          ]),
+
+      // identifier: cooking state, represent by `OvenStates.cooking`.
+      State<OvenStates>(OvenStates.cooking,
+          // events:
+          // When the button `paused` is pressed during cooking state,
+          // OvenState will changed to `OvenStates.paused` state.
+          //
+          // When the button `counter` time is elapsed during cooking state,
+          // OvenState will changed to `OvenStates.completed` state.
+          events: {
+            Logic(name: 'button_pause')..gets(button.eq(Button.pause())):
+                OvenStates.paused,
+            Logic(name: 'counter_time_complete')..gets(counter.val.eq(4)):
+                OvenStates.completed
+          },
+          // actions:
+          // During the cooking state, `led` is change to yellow; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 1 (Enable value update).
+          actions: [
+            led < LEDLight.yellow().value,
+            counterReset < 0,
+            en < 1,
+          ]),
+
+      // identifier: paused state, represent by `OvenStates.paused`.
+      State<OvenStates>(OvenStates.paused,
+          // events:
+          // When the button `resume` is pressed during paused state,
+          // OvenState will changed to `OvenStates.cooking` state.
+          events: {
+            Logic(name: 'button_resume')..gets(button.eq(Button.resume())):
+                OvenStates.cooking
+          },
+          // actions:
+          // During the paused state, `led` is change to red; timer's
+          // `counterReset` is set to 0 (Do not reset);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.red().value,
+            counterReset < 0,
+            en < 0,
+          ]),
+
+      // identifier: completed state, represent by `OvenStates.completed`.
+      State<OvenStates>(OvenStates.completed,
+          // events:
+          // When the button `start` is pressed during completed state,
+          // OvenState will changed to `OvenStates.cooking` state.
+          events: {
+            Logic(name: 'button_start')..gets(button.eq(Button.start())):
+                OvenStates.cooking
+          },
+          // actions:
+          // During the start state, `led` is change to green; timer's
+          // `counterReset` is set to 1 (Reset value);
+          // timer's `en` is set to 0 (Disable value update).
+          actions: [
+            led < LEDLight.green().value,
+            counterReset < 1,
+            en < 0,
+          ])
+    ];
+
+    // Assign the oven StateMachine object to public variable declared.
+    oven = StateMachine<OvenStates>(clk, reset, OvenStates.standby, states);
+  }
+
+  // An ovenStateMachine that represent in getter.
+  StateMachine<OvenStates> get ovenStateMachine => oven;
+}
+
+void main() async {
```

Comment:
Maybe use `Future<void> main ...` like in the other examples?

Reply:
Ya, can do that.